ResearchKit/ActiveTasks/ORKActiveStepQuantityView.m

```diff
@@ -111,91 +110,89 @@ - (void)setImage:(UIImage *)image {
                                                                   attribute:NSLayoutAttributeBottom
                                                                  multiplier:1.0
                                                                    constant:0.0]];
-    [additionalConstraints addObject:[NSLayoutConstraint constraintWithItem:_valueHolder
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:_valueHolder
                                                                   attribute:NSLayoutAttributeLeft
                                                                   relatedBy:NSLayoutRelationGreaterThanOrEqual
                                                                      toItem:self
                                                                   attribute:NSLayoutAttributeLeft
                                                                  multiplier:1.0
                                                                    constant:0.0]];
-    [additionalConstraints addObject:[NSLayoutConstraint constraintWithItem:_valueHolder
+    [constraints addObject:[NSLayoutConstraint constraintWithItem:_valueHolder
                                                                   attribute:NSLayoutAttributeRight
                                                                   relatedBy:NSLayoutRelationLessThanOrEqual
                                                                      toItem:self
                                                                   attribute:NSLayoutAttributeRight
                                                                  multiplier:1.0
                                                                    constant:0.0]];
-    for (NSLayoutConstraint *constraint in additionalConstraints) {
-        constraint.priority = UILayoutPriorityRequired-2;
+    for (NSLayoutConstraint *constraint in constraints) {
+        constraint.priority = UILayoutPriorityRequired - 2;
     }
     
-    [NSLayoutConstraint activateConstraints:additionalConstraints];
-    
+    [NSLayoutConstraint activateConstraints:constraints];
     
-    NSLayoutConstraint *zeroWidthConstraint = [NSLayoutConstraint constraintWithItem:self
-                                                                           attribute:NSLayoutAttributeWidth
-                                                                           relatedBy:NSLayoutRelationEqual
-                                                                              toItem:nil
-                                                                           attribute:NSLayoutAttributeNotAnAttribute
-                                                                          multiplier:1.0
-                                                                            constant:0.0];
-    zeroWidthConstraint.priority = UILayoutPriorityRequired-1;
-    _zeroWidthConstraint = zeroWidthConstraint;
-    _zeroWidthConstraint.active = !_enabled;
```

Comment:
Activation of the `_zeroWidthConstraint` constraint is taken care in `updateConstraints`, which will take place at the most optimal time after `setNeedsUpdateConstraints` is called.